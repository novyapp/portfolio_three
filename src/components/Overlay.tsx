import React, { forwardRef } from "react";

const Overlay = forwardRef<HTMLDivElement>((_, ref) => {
  return (
    <div className="absolute left-10 top-10" ref={ref}>
      <h1 className="bg-gradient-to-r from-orange-400 via-pink-500  to-pink-700 bg-clip-text text-4xl font-extrabold  uppercase text-transparent">
        Michal
        <br /> Nowotnik
      </h1>
      <h2 className="text-md uppercase text-white">Front-End Developer</h2>
    </div>
  );
});

export default Overlay;
