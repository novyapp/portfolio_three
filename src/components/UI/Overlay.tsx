import React, { createRef, RefObject, useEffect, useRef } from "react";
import Buttons from "./Buttons";
import Header from "./Header";
import SocialLinks from "./SocialLinks";
import { gsap } from "gsap";

interface OverlayProps {
  gotoSection: (value: number) => void;
  canvasAnim: RefObject<HTMLCanvasElement>;
}

function Overlay({ gotoSection, canvasAnim }: OverlayProps) {
  console.log(canvasAnim);
  const headerRef = createRef<HTMLDivElement>();
  const socialRef = createRef<HTMLDivElement>();
  const menuRef = createRef<HTMLDivElement>();

  useEffect(() => {
    const tl = gsap.timeline({ defaults: { duration: 1 } });
    tl.fromTo(canvasAnim.current, { y: "0%" }, { y: "0%" }).duration(1);
    tl.fromTo(headerRef.current, { y: "-140%" }, { y: "0%" }).duration(1.5);
    tl.fromTo(socialRef.current, { y: "180%" }, { y: "0%" }).duration(1.8);
    tl.fromTo(menuRef.current, { x: "-180%" }, { x: "0%" }).duration(2);
  }, []);

  return (
    <>
      <Header ref={headerRef} />
      <Buttons gotoSection={gotoSection} ref={menuRef} />
      <SocialLinks ref={socialRef} />
    </>
  );
}

export default Overlay;
