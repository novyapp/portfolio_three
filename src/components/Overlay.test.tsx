import React from "react";
import Overlay from "./Overlay";
import { render } from "@testing-library/react";

describe("Overlay", () => {
  it("renders correctly", () => {
    const { container } = render(<Overlay />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('has a "Michal Nowotnik" h1 tag', () => {
    const { getByText } = render(<Overlay />);

    expect(getByText(/Michal/)).toBeInTheDocument();
    expect(getByText(/Nowotnik/)).toBeInTheDocument();
  });

  it('has a "Front-End Developer" h2 tag', () => {
    const { getByText } = render(<Overlay />);

    expect(getByText(/Front-End Developer/)).toBeInTheDocument();
  });
});
