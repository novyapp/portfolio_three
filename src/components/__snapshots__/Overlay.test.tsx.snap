// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay renders correctly 1`] = `
<div
  class="absolute left-10 top-10 "
>
  <h1
    class="bg-gradient-to-r from-orange-400 via-pink-500 to-pink-700 bg-clip-text text-4xl font-extrabold uppercase text-transparent"
  >
    Michal
    <br />
     Nowotnik
  </h1>
  <h2
    class="text-md uppercase text-white"
  >
    Front-End Developer
  </h2>
</div>
`;
